File: app/src/main/java/com/example/qrlogin/ToolbarMenuHandler.java
package text.only.app.qrlogin;

import text.only.app.R;
import text.only.app.ScanQRActivity;
import android.app.Activity;
import android.content.Intent;
import android.widget.ImageButton;
import androidx.appcompat.widget.Toolbar;
import android.widget.PopupMenu;

public class ToolbarMenuHandler {
    public static void setupToolbar(Activity activity, Toolbar toolbar, ImageButton btnSettings) {
        btnSettings.setOnClickListener(v -> {
            PopupMenu popup = new PopupMenu(activity, btnSettings);
            popup.getMenuInflater().inflate(R.menu.menu_toolbar, popup.getMenu());
            popup.setOnMenuItemClickListener(item -> {
                if (item.getItemId() == R.id.action_profile) {
                    // Deschide activitatea de profil
                    activity.startActivity(new Intent(activity, ProfileActivity.class));
                    return true;
                } else if (item.getItemId() == R.id.action_add_device) {
                    // Deschide activitatea de scanare QR
                    activity.startActivity(new Intent(activity, ScanQRActivity.class));
                    return true;
                }
                return false;
            });
            popup.show();
        });
    }
}

File: app/src/main/java/text/only/app/qrlogin/ToolbarMenuHandler.kt
package text.only.app.qrlogin

import android.content.Intent
import android.widget.ImageButton
import android.widget.PopupMenu
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.Toolbar
import text.only.app.OnlyGuardActivity
import text.only.app.ProfileActivity
import text.only.app.R

object ToolbarMenuHandler {
    fun setupToolbar(activity: AppCompatActivity, toolbar: Toolbar?, btnSettings: ImageButton) {
        btnSettings.setOnClickListener {
            val popup = PopupMenu(activity, btnSettings)
            popup.menuInflater.inflate(R.menu.menu_toolbar, popup.menu)

            popup.setOnMenuItemClickListener { item ->
                when (item.itemId) {
                    R.id.action_profile -> {
                        activity.startActivity(Intent(activity, ProfileActivity::class.java))
                        true
                    }
                    R.id.action_only_guard -> {
                        activity.startActivity(Intent(activity, OnlyGuardActivity::class.java))
                        true
                    }
                    else -> false
                }
            }
            popup.show()
        }
    }
}
